### PR TITLE
Remove (allocation,blobber) from challenge partition if blobber is empty now

### LINKED ISSUE #646: Remove challenges for deleted files.
WrtieMarkers need to be sent to chain when we `add` a new file, `update` a file, `delete` a file, `copy` or `move` a file, and `rename` a file.

0chain needs to make appropriate changes to challenges. For instance, removing challenges for deleted files. Some of this already works, for example when a file is added. We need to fill in the blanks.

This needs to be coordinated with a matching change to the blobber.

Any crud action that changes the tree is reported to the chain because the Merkle root of the allocation changes as a result

### PR DESCRIPTION
**Pull request by ssardana08**
## Changes
- remove allocation from blobber_allocation_challenge partition, if blobber.UsedSize is now empty

## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/smartcontract/storagesc/blobber.go [hunk 1]
```diff
@@ -685,7 +685,13 @@ func (sc *StorageSmartContract) commitBlobberConnection(
 	}
 
 	// the first time the allocation is added  to the blobber, created related resources
-	if blobAlloc.BlobberAllocationsPartitionLoc == nil {
+	if blobAlloc.Stats.UsedSize == 0 {
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/storagesc/blobber.go:673]: Isn't the UsedSize incremental only? so this is actually checking if the blobber is redeeming a read marker that its allocation had no data been written to it yet, is that possible?

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/storagesc/blobber.go (hunk 1 above)
Comment (ssardana08) [reply code/go/0chain.net/smartcontract/storagesc/blobber.go:673]: it's not always incrementing...
`blobAlloc.Stats.UsedSize += commitConnection.WriteMarker.Size`
even though it's a +=, but on deletion of file Writemarker.Size is in negative which will reduce the usedSize

Comment (peterlimg) [reply code/go/0chain.net/smartcontract/storagesc/blobber.go:673]: @cnlangzi please confirm that if the WriteMarker.Size could be negative.

Comment (cnlangzi) [reply code/go/0chain.net/smartcontract/storagesc/blobber.go:673]: @sum-it08 @peterlimg  WriteMaker.Size can be any value( >0,0,<0). eg
- upload/update: >0
- delete: <=0
- rename: 0

**Review by peterlimg — APPROVED** [review]
lgtm

**Pull request merged**